lib/puppet/resource/status.rb

```diff
@@ -167,11 +171,13 @@ def initialize(resource)
         @time = Time.now
         @events = []
         @resource_type = resource.type.to_s.capitalize
+        @provider_used = resource.provider.class.name unless resource.provider.class == NilClass
```

Comment:
I was thinking we'd include the stringified `name` of the provider, e.g. resource.provider.name.to_s, not the provider's class name. The former is what the provider author registers (and what the manifest author can specify to pick an explicit provider), while the latter is some what magical. For example, these are the known package providers:

```
 Puppet::Type.type(:package).providers
=> [:aix, :appdmg, :apple, :dpkg, :apt, :aptitude, :rpm, :aptrpm, :sun, :blastwave, :yum, :dnf, :fink, :openbsd, :freebsd, :gem, :hpux, :macports, :nim, :opkg, :pacman, :pip, :pip3, :pkg, :pkgdmg, :pkgin, :pkgng, :pkgutil, :portage, :ports, :portupgrade, :puppet_gem, :rug, :sunfreeware, :tdnf, :up2date, :urpmi, :windows, :zypper]
```

But the class names are automagically generated, e.g. `Puppet::Type::Package::ProviderYum`

Reply:
The `unless` part can be shorted to just `unless resource.provider.nil?`

Reply:
We might want to look at creating a ticket to update some error messages to use provider.name instead of the class name. When I was poking around the transaction I saw that we were using the class name in at least one user-facing message.

Reply:
I'd started with resource.provider.name.to_s, but wasn't sure that was what we wanted, I'll switch it back